Feat/add_pages

suchachevskaya opened the pull request:
- Основные компоненты проекта: роутинг, базовые страницы (Favorites, History, Login, Registration).
- Реализован глобальный Header для отображения на всех страницах.
- Структура проекта организована для удобства дальнейшей разработки.

suchachevskaya committed:
header

Разработка шапки сайта с навигацией и роутингом

suchachevskaya committed:
Рефакторинг роутинга

Рефакторинг роутинга для устранения повторения Header:
- Переместила Header для глобального отображения на всех маршрутах
- Убрала дублирующийся код Header в каждом маршруте
- Улучшена структура проекта для удобства поддержки

suchachevskaya committed:
react 18

ElfistIO commented on src/App.tsx:
```diff
@@ -1,35 +1,57 @@
+      <Header />
+      <Routes>
+        <Route
+          path="/"
```
пути в константу по типу ROUTES = { HOME: '/', ... }

ElfistIO commented on src/App.tsx:
```diff
@@ -1,35 +1,57 @@
+        <Route
+          path="/"
+          element={
+            <>
```
А зачем тут везде фрагменты? Не нужны

ElfistIO commented on src/favorites/Favorites.tsx:
```diff
@@ -0,0 +1,8 @@
+import "../history/history.css"
+export default function History() {
```
Не используем экспорт по дефолту

ElfistIO commented on src/favorites/favorites.css:
```diff
@@ -0,0 +1,22 @@
+  justify-content: space-between;
+  width: 100%;
+  height: 10%;
+  background-color: #404040;
```
цвета вынесите в переменные в отдельный файл и потом используйте где нужно

ElfistIO commented on src/header/Header.tsx:
```diff
@@ -0,0 +1,24 @@
+          <img src="src/header/icon_header/logo-museum.png" alt="Logo" />
+        </figure>
+      </Link>
+      <nav className="header__navigation">
```
создайте объект с необходимыми полями и бегайте по нему map'ом

suchachevskaya committed:
Рефакторинг навигационной логики и оптимизация маршрутизации

- Создан отдельный файл для маршрутизации.
- Заменены экспорты по умолчанию на именованные экспорты.
- Реализован массив объектов navItems с полями id, text, to и modifier.
- Навигационные элементы динамически генерируются с помощью метода map.
-Создан общий SCSS-класс для переменных.

suchachevskaya committed:
Merge branch 'develop' into feat/addPages

suchachevskaya committed:
Merge branch 'develop' into feat/addPages

ElfistIO commented on src/header/Header.tsx:
```diff
@@ -0,0 +1,42 @@
+import { Link } from "react-router-dom";
+import "../header/header.css";
+import { ROUTES } from "../routes";
+const navItems = [
```
давайте подобные вещи выносить в отдельный модуль constants.ts рядом с компонентом
